docs/data/tree-view/rich-tree-view/customization/FileExplorer.js

```diff
@@ -218,20 +213,19 @@ const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
     getGroupTransitionProps,
     getDragAndDropOverlayProps,
     status,
-    publicAPI,
   } = useTreeItem({ id, itemId, children, label, disabled, rootRef: ref });
 
-  const item = publicAPI.getItem(itemId);
-  const expandable = isExpandable(children);
+  const item = useTreeItemModel(itemId);
+
   let icon;
-  if (expandable) {
+  if (status.expandable) {
     icon = FolderRounded;
   } else if (item.fileType) {
     icon = getIconFromFileType(item.fileType);
   }
 
   return (
-    <TreeItemProvider itemId={itemId}>
+    <TreeItemProvider {...getContextProviderProps()}>
```

Comment:
This is a very dumb BC...
I need the id attribute in `TreeViewChildrenItemProvider` to be able to generate the defaultized id attribute on the 1st render?

Why?
Because `state.items.itemMeta` is empty on the 1st render, before this PR it was not a problem because the item was rendering twice (because of lack of memoization) so we would get the item meta on the 2nd render.
But now we only have one render and we can't access the `itemMeta` from the state.
So we need another way to access the id attribute manually passed to `TreeItem` in `SimpleTreeView`.

I should have created `getContextProviderProps` from the beginning "just in case"